### PR TITLE
Add Ruby 2.6/2.7 support

### PR DESCRIPTION
**Pull request by bastelfreak**
This is a resubmission of https://github.com/voxpupuli/json-schema/pull/437

### CODE REVIEW

File: lib/json-schema/attributes/formats/ip.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def self.validate(current_schema, data, fragments, processor, validator, options
         begin
           ip = IPAddr.new(data)
         rescue ArgumentError => e
-          raise e unless e.message == 'invalid address'
+          raise e unless e.message.start_with?('invalid address')
         end
```
Comment (ekohl) [inline lib/json-schema/attributes/formats/ip.rb:15]: I feel that parsing exceptions is fragile and I'd implement it like this (roughly):
```ruby
class IPFormat < FormatAttribute
  def self.validate(current_schema, data, fragments, processor, validator, options = {})
    return unless data.is_a?(String)

    unless self.regex.match?(data)
      error_message = ....
      validation_error(...)
    end
  end

  def self.regex
    raise NotImplementedError
  end
end

class IP4Format < IPFormat
  def self.ip_version
    4
  end

  def self.regex
    Resolv::IPv4::Regex
  end
end
```

I do think `regex.match?()` may be Ruby 2.5 only but you can use `=~` for older Rubies.

However, I'm not sure if this regex is available in older Rubies.

**Comment by os6sense** [comment]
Any plans to deal with the deprecation warnings that are raised when using 2.7.x?  e.g. 

```
/home/runner/work/M3/M3/vendor/bundle/ruby/2.7.0/gems/json-schema-2.8.1/lib/json-schema/schema/reader.rb:121: warning: calling URI.open via Kernel#open is deprecated, call URI.open directly or use URI#open
```

**Comment by bastelfreak** [comment]
@os6sense we do plan to fiy those in the future when we've some time. I also havent reviewed the remaining PRs yet. Feel free to take a look as well or zo provide a new PR that fixes this. every contribution is welcome!

**Comment by bastelfreak** [comment]
@ekohl are you fine with merging this for now and update it later? Or can you submit your code as a PR?

**Review by ghoneycutt — APPROVED** [review]

**Pull request merged**